internal/api/router_test.go

```diff
@@ -0,0 +1,167 @@
+package api
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type want struct {
+	code        int
+	response    string
+	contentType string
+}
+
+type testcase []struct {
+	name   string
+	url    string
+	method string
+	want   want
+}
+
+var router = SetupRouter()
+
+func Test_update(t *testing.T) {
+
+	tests := testcase{
+		{
+			name:   "service_post_200_gauge",
+			url:    "/update/gauge/someMetric/23.4",
+			method: http.MethodPost,
+			want: want{
+				code:        200,
```

Comment:
Можно использовать коды вместо чисел:
```suggestion
				code:        http.StatusCreated,
```